lib/streamlit/app_session.py

```diff
@@ -129,8 +124,21 @@ def __init__(
 
         self._session_state = SessionState()
 
+        self.register_change_listeners()
+
         LOGGER.debug("AppSession initialized (id=%s)", self.id)
 
+    def register_change_listeners(self):
+        self._local_sources_watcher.register_file_change_callback(
+            self._on_source_file_changed
+        )
+        self._stop_config_listener = config.on_config_parsed(
+            self._on_source_file_changed, force_connect=True
+        )
+
+        # The script should rerun when the `secrets.toml` file has been changed.
+        secrets._file_change_listener.connect(self._on_secrets_file_changed)
```

Comment:
It looks like this function is only called from the constructor. Assuming I'm not missing another call site, is there value to having it broken into a separate function as opposed to inline?

Reply:
No strong reason for having it this way, it is mostly because the separate method was called elsewhere in previous versions of the code and I didn't inline it when I reverted to the constructor doing it. I could see a case for inlining it (no pointless indirection) or keeping it separate (pretty clearly related chunk of code can benefit from having a name)

Reply:
I think it's a personal call, then. But if it's just internal, it should probably be named `_register_change_listener` instead